java/pom.xml

```diff
@@ -658,6 +658,19 @@
         <module>gandiva</module>
       </modules>
     </profile>
+
+    <profile>
+      <!-- to speed up the docker-compose builds -->
```

Comment:
Instead of doing this, can you just map a directory where the local maven repo is? The downloading takes the vast majority of the time and if you can a repo outside of the image, it runs pretty quick.  For instance running docker with this command mapping my local maven repo was able to build and test in less than 30s

`docker run -v $HOME/.m2:/root/.m2 arrow-java`

``` Reactor Summary:
[INFO] 
[INFO] Apache Arrow Java Root POM ......................... SUCCESS [  1.967 s]
[INFO] Arrow Format ....................................... SUCCESS [  2.158 s]
[INFO] Arrow Memory ....................................... SUCCESS [  2.158 s]
[INFO] Arrow Vectors ...................................... SUCCESS [  6.108 s]
[INFO] Arrow Tools ........................................ SUCCESS [  4.656 s]
[INFO] Arrow JDBC Adapter ................................. SUCCESS [  1.890 s]
[INFO] Arrow Plasma Client ................................ SUCCESS [  0.147 s]
[INFO] Arrow Flight ....................................... SUCCESS [ 10.481 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 29.825 s
```

Reply:
My idea was to both download the deps and build the java libs under `docker_cache` (similarly like the cpp docker setup works) in order to prevent contaminating the local development environment - I guess in case of java We don't need that. 

BTW What machine do You build on? My total build time is 01:35 min.

Reply:
> My idea was to both download the deps and build the java libs under docker_cache (similarly like the cpp docker setup works) in order to prevent contaminating the local development environment - I guess in case of java We don't need that.

I think its fine to reuse the local Maven repository for dependencies, but we probably don't want to publish our artifacts there, which would contaminate it.

> BTW What machine do You build on? My total build time is 01:35 min.

My laptop is a newer I7 8 core with 32G ram and running Ubuntu - not too sure whats making that much of a difference though..